### PR TITLE
[SouthPark] South Park with Spanish Latam (Regex Fixed)

### PR DESCRIPTION
**Pull request by darkxex**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Support for South Park Latam....

note: I can't shorten the regex any more, without breaking the script.

### CODE REVIEW

**Comment by darkxex** [comment]
@pukkandan could you check now please? now the amount in the regex is smaller and it's working.

**Comment by pukkandan** [comment]
See my comment on #3831

**Comment by darkxex** [comment]
> Comment on #3831 still apply

I just don't understand friend, what's wrong. it's almost the same as what you put here, only what you put breaks the script.
image

i edited to:
_VALID_URL = r'https?://(?:www\.)?(?P<url>southpark\.lat/(?:en/)?(?:video-?clips?|collections|episod(?:e|io)s)/(?P<id>(?P<unique_id>.+?)/.+?)(\?|#|$))'

the change I made was minimal from what you gave as an example, if you accepted the southpark.de which is literally the same thing. just change the domain and "episodes" to "episodios".
the only thing that occurs to me is that they don't want to accept it for some racist reason against Spanish speakers or something, because I don't really see the point of it.

File: yt_dlp/extractor/southpark.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def _get_feed_query(self, uri):
 
 class SouthParkLatIE(SouthParkIE):
     IE_NAME = 'southpark.lat'
-    _VALID_URL = r'https?://(?:www\.)?(?P<url>southpark\.lat/(?:en/)?(?:video-?clips?|collections|episod(?:e|io)s)/(?P<id>(?P<unique_id>.+?)/.+?)(\?|#|$))'
+    _VALID_URL = r'https?://(?:www\.)?(?P<url>southpark\.lat/(?:en/)?(?:video-?clips?|collections|episod(?:e|io)s)/(?P<id>(?P<unique_id>.+?)/.+?))'
```
Comment (pukkandan) [inline yt_dlp/extractor/southpark.py:114]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?southpark\.lat/(?:en/)?(?:video-?clips?|collections|episod(?:e|io)s)/(?P<id>[^/?#&]+)'
```

File: yt_dlp/extractor/southpark.py [hunk 2]
```diff
@@ -109,6 +109,48 @@ def _get_feed_query(self, uri):
...
+            'ext': 'mp4',
+            'title': 'South Park',
+            'description': 'md5:ae0d875eff169dcbed16b21531857ac1',
+        },
+    }]
+
+    def _get_feed_url(self, uri, url=None):
+        video_id = self._id_from_uri(uri)
+        config = self._download_json(
+            'http://media.mtvnservices.com/pmt/e1/access/index.html?uri=%s&configtype=edge&ref=%s' % (uri, url), video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/southpark.py:147]:
```suggestion
        config = self._download_json(
            f'http://media.mtvnservices.com/pmt/e1/access/index.html?uri={uri}&configtype=edge&ref={url}',
            video_id)
```

**Comment by pukkandan** [comment]
if your response to asking for changes is to call me a racist, you better take your PR elsewhere

**Comment by pukkandan** [comment]
Let me know if anything is broken rn

**Comment by darkxex** [comment]
sorry for calling you racist, i even deleted it because i realized it was dumb of me.
I think it works fine.

**Pull request merged**